### ISSUE TITLE
feat: add Starport for better development environment

### ISSUE THREAD

**Issue by jaybxyz** [tendermint/farming#203]
## Summary

Right now, the farming module uses [localnet](https://github.com/tendermint/farming/blob/main/scripts/localnet.sh) script to run local network for testing local environment. This issue proposes to add Starport for better development environment so that just by running `starport chain serve` command, you can run local network with various different functionalities that Starport supports. 

## Tasks

- Add `config.yml` to add Starport
- Update `README.md`and `TECHNICAL_SETUP.md` docs
- Remove `scripts/localnet.sh` and update `Makefile`
- Remove most of the `third_party` proto files 

## References

- [Starport Official Github](https://github.com/tendermint/starport)
- [Starport Documentation](https://docs.starport.com/)

____

#### For Admin Use

- [x] Not duplicate issue
- [x] Appropriate labels applied
- [x] Appropriate contributors tagged
- [x] Contributor assigned/self-assigned

**Comment by kogisin** [comment]
The farming module is planning to release the next version tomorrow. I think this PR is better not to be merged in the next version since I don't have strong confidence at this point. As I integrate Starport into the module, I find out there are tradeoffs (e.g proto-format functionality when `starport g proto-gen` and there should exist scaffold inline comments in order to fully utilize Starport subcommands and etc). It might be because I don't know there is a workaround or there are some stuff that I don't know yet. Therefore, I think I need to spend more time playing with it.

**Comment by dongsam** [comment]
> The farming module is planning to release the next version tomorrow. I think this PR is better not to be merged in the next version since I don't have strong confidence at this point. As I integrate Starport into the module, I find out there are tradeoffs (e.g proto-format functionality when `starport g proto-gen` and there should exist scaffold inline comments in order to fully utilize Starport subcommands and etc). It might be because I don't know there is a workaround or there are some stuff that I don't know yet. Therefore, I think I need to spend more time playing with it.

I agree, Let's continue it after starport starts supporting cosmos-sdk 0.44.3
